<!DOCTYPE html>
<html lang="en">

<head>
	<title>Binning Mitochondrial Reads From Skimming Data Using Krepp | Ali Osman Berk Şapcı</title>
	<link rel="canonical" href="https://bo1929.github.io/">
	<link rel='alternate' type='application/rss+xml' title="Ali Osman Berk Şapcı RSS" href='/index.xml'>
	<link rel='stylesheet' type='text/css' href='/style.css'>
	<link rel="icon" href="favicon.ico">
	<meta name="description" content="Binning mitochondrial reads from skimming data using krepp and all mitogenomes available on RefSeq.">
	<meta name="keywords" content="k-mers, metagenomics, mitochondrial-reads, distance-estimation, krepp, genome-skimming, read-mapping">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="robots" content="index, follow">
	<meta charset="utf-8">
</head>


<link rel="preload stylesheet" as="style" href="https://bo1929.github.io/syntax_light.min.css" />
<link rel="preload stylesheet" as="style" href="https://bo1929.github.io/syntax_dark.min.css" />

<div class="js-toggle-wrapper">
    <div class="js-toggle">
        <div class="js-toggle-track">
            <div class="js-toggle-track-check">
                <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAABlJJREFUWAm1V3tsFEUcntnXvXu0tBWo1ZZHihBjCEWqkHiNaMLDRKOtQSKaiCFKQtS/SbxiFCHGCIkmkBSMwZhQNTFoQZD0DFiwtCDFAkdDqBBBKFj63rvdnfH7zfVo5aFBj0l2Z/dm5vd98/0es8dYjlpr62azufnDQNZcU1PciMfjWvb9rvZSMk4Ayfb36pLH13189GC8LAtIRLLPt+pzwrCuLq4ISEv/gHmitrAwfPbEkXc/ad4dL6iujrvyX0jcitgd/yZlZqftP6995Mr5TVLa22Tn8XVX2g/XLSRjUu7Q79jonS7I7hS7/0oOb5VyqF52n98oj7esXX07EjlxwXWisRmSnm3b29TTM8iYrjmFBWExubxwY/uhNas4r/WySl1fc5cetDMd7ydl+lMJJRw5WC8ud62Xx5rfepzwxgZmbhUYNS5Stvsj4yo2GXJEFBVHWDBkfdbR9HpYBaaUajDnBLKKpl1xRKYcgGtMCqEzTaSnThk/SQT0uJqTqFNBmXMCsZE48DzRZRMBRjv1GHNdk3HBImF9ZUvTyxM40pMKVc4JZBXQOLOFoDeKSxdp6HIQcO4rjYT9fn0pjbz9GLt7BAAODmjSVReXUMFzNW5x5vfxp2mIxZjIuQKJxAmFa+is2DQJJQ0JyBVExNOYcJnPxx/6/utnijmP555ALEagKAGGnGn64QORBjARcIA/yJk7JMJBLRrNtybTvH88KGjCf2jK86bhzmMcwDKFZEQvbIhxFYhChoMWMzU2iWznlIBEVJOsP+1bdX/ALx9l7jApADeDAEcMkE90JnUmmGl4USKQ0xhoW3JB5XY0YrxYWhLwMZZypUyjDGH35AbNwgUGiFBPpuGbHCpAOV1ZGXf2f/taftAv31DyeymN2d1IhAFAwTOmnzF/kKcdh3me7CYCOVNgycju84u8DeVlwfFq9/ZlTfldYrMUjOlrkjkD+rU+WzCROkcEchIDHR011syZW9JHD7y07N6JvhWMpz3pugaTkB6lWFVCKkhck0zzeMp2utq+uHrmfxOgoCO/Z8CXPlEQ1bdH8wgvhSIkEG0ICcQeExIFGdimjvKka7btJFZuaXOammIGKUCFQ53j9EN1dYKWqHf0t2w407W2tgs6h89ZnImjB55flh81tt9XirjjDuSl+oIPRQ0iWPgNZ5GqTqbBe3vSzEl5n5PhWKwocyR2HlqYN61qV18WjYjE8JLARZPQsUSim8foIRYTlGr02Ly7piASFRtKJ4VfieYhxdS2JcDVMN6xVOKZyrCGm8b108lrLRVzvptLH7IoEFLFANes6KnDi+uxfmvFnF17oALq5u1agu3/YfHkcSFzeSggV5eXRfIB7CHNcO5SUI+Ih5Ir7f4MAV9IqdFzdZgNpZw1Gcs1mNvgGbTbqQ9/cz7ZuuhgyYRQ49ljTyWHhr2DwpNHHFf+5gnWZ3Bharo+0TD5dNMw5vv9RlVpSRDHK4TlnoukhtYApuOHejSZQuo5g/A9BysdKRCyLl6062fN37OXMDlvUJtUrtmxo0avrW3wTrYs3jJ9RvRVChrmSmanPMpX2OXMsmDGh6AiEIwBAlvkOqIdBy+8JyAz8pz7QxiDth4KDy5uAlwzrWTnwC8Vc4KVAMZ3YUZ+IqoIjP3h5KFFX1ZMy3uW+7RhEDHgTi0zC9rS7uhPCDiNrGFyqBeERtKN/B0YlyFCkw0NJ5C0Ojv7zvT1a1WV1TuvZDdL4NTgB7CASYpsen6gqvG5jmTf5qHedADgkBl3D0nkSgNhZACDyi0FUKZRr3IdRjgN4WPPoFMIIegIK3mqd38fS80mcJKelM4szNyzZtQbkchGePuBRS8Eg9pHU8ojRQpSqs+ajAIwTjjUMQ/nvTNM0kicwYxZIYMh/891DYi+fvedB+c1xsm4lDU6ya+Axtz+RiAzEVYbajQOpq17F0R9QevNcEhfcU+xvyQQUalGJBSesqOkgPQ4YNyUZL9fSvUPDjoNAwN8/dwFjaczNkc3ptaMud1EIDtGcmXTcefO2cGSvKIFfp/2JIJxlq7xEl3nVPM4fDeIbPkD16/ptNc0bDu7qxbsu0R2JGywWMIjF2ft3tjfloAyQAGXiOn8hrqwbVvMXzaO+QeHXP6nF0wvX74Hf4NGG5GPjSlYoyM3P/0FbCT6zvM/yYoAAAAASUVORK5CYII=" role="presentation" style="pointer-events: none;" width="16" height="16">
            </div>
            <div class="js-toggle-track-x">
                <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAABwNJREFUWAmtV1tsFFUY/s6Z2d22zLYlZakUCRVaQcqlWIiCiS1gTEB9UAO+GR9En3iQGI0xJiSiRB98MjEq8cEQTSBeHhQM0V7whtEGDWC90BYitxahtNtu25058/v/ZzvLbilawJNM5+yZ89+//1LgJhYRNLW1uDfBAvpGiIk2O5auvfFxqIH3ZJ8/u06GN6Z9+wVl5SjcD1IbZa/UPkPyYl2uR4dreoD2bnbYxTlBBRytkHXtAREphP5KuH4lddx9h70yxX05t7yYXwGb6W8nx1jibpl2rFlGBxcG9M18okOrn7Bnk/BAO/4bI0UeEE1zjBp3UmvjOxJXJdaKN/ZiIu4tOZrAb4aTdZAZArKmWeiiJZ6jt5tiagdCS9+6cgO1Ne6Mvhe+ixTIfyDVhipnK9p+P0Edqx9RW/YZtQVGmOLChRxNNlyPsTEgPQKMB3dbEHa0h1awYmQ83enTd2vmUtvKd1Glv2RkzBb+kZGRrKtjzG60Wguhd/lJZBingbcfWWe72vjT75bJDrhYtvA0hrurETDr5HyF2Knb1MM4ab//xIoOqueA0edRnkkinTyJdYvqLFDZO4zUPFCvVoDjJq4T7TE61IWh4x5KqxX5KVKkX8WZ/t2ov2cb3MHt4dhIyOxIJxJOOF6xRx/99BksXLoecWcXytILMNBDqKpnGZWPquYfPxY8iXGR9fK+SgFrgcRPXPjVqhehL+3EmZ5RGJQi1QBU8TPThQnOQzm+5UXGIcetUeEAfP13VwzpI+w1jGJWdSliNfvVhiMPiOsllJag4M/UGHiqM6dlBb2OTLKHHV6KkvogrJ4XhBWniWK/Gp1MQyf93FOeUXKmKk/FzJxbQtKLjFXYT4USupy8fQVir2ynVEBiZMG0qtOHMS/AW4Gwrk7BG3C1F0B5nqNKE0CME4MfVRLPnXkBKe+ipvoFhNQywOhdghvLi0F8ReyVXV4BKTBRbbe5f64zR/DHsdZw1hJfeWlHl/GNRJzDxrd5m192z78TMaVnKELZoINZS4BzQ7vtnZljSnha/pPCbkuxzXcupYwI5tIeCpGc0Yp9tWHZQy/rmYhRfNgg4bHJBYLzGkxsRJF4XKlE2jBOHNSv3kY7Tj6vthzPFl61BrYwqFlmEQhtSVXmLiksxLmtRgYXI1ULU61JJ4eVKmG3/5sCVgpbMT6OMJ2E08/29Xf3w6v4FnHdCjfWgXu/O8Z5mLdCkeRs2khHe1DqOtQwbHWTAnM5S2HNmhALYo5KjkPFrMMKjZl6HxhWIAb0BqE+/73GrBRQUsKYiBu4JX8ycI6wtw+i5ef3NZpsrKVSHYCP37jwGDgeE1SA0S/xtl5SU2fs1ApEp0qTLVRjgyycDSsLHMSwmFltZMStR3uLLg6BdLhDa5dC6ryU2pHBe1BVO9tUcwfitJt2CLJZUHoG6T7Op75u0IyK31TCPcwFqgPk/KCaD3dFOuZBCO7xvCT/j048b3I3c7F2+WuOW7qdgkucFYlcQ4qop3yzTX7WaKfOCccye3Ts1Etq0+a/BHCF1yPgF3tAUkR6OrtGmo6gl94qqcXKh3rDyrOkPa58URoWcov2Mo6M+0QjrqKB+b7++oMa9Sz+ZkM0mie6aAtnGUvhmxaI+TogPOSQedgWioGSHFLn3v4kLh4HRspNmOGv41k+55siLFp2z6xYeJjhljFcbmxJlr4ga06TbevSByz/glQq4BJx46/c+237PbBqEYKxX3HpmKZEnQnr65X20hqJYaNcLoFOLiJk2LuBbyg7Q0OEn+hm0P3honxFD6rdxYorKpeIoi4YSSvyQHQIbM5t4+YNxLj/OxhVOOE4585qGpjnq+wSx6Q9CtNxTjd5klB+g6Mv36r0+b9cZFi44WYkHdG2ZWb3TtOUOXyVAlKlpGvJIAJ3eBMyfYS5C0qRZGtC85j+4sOasDe9xznPYezhhO/2Q6eP2fSOvYHOjtuQ1a9Q1VKynVDaMc8E0tptdxUsTFpFIYjcZKcbnoaQTNdiqCwNlL4G7oziSqGnT1ALf34vhk4R5zU3qYV9ONp9K88RtouShE68JwaU8dFw5W617shWa9ykeaBIn2hcsvPgL00k45QdTCZuSVcTRNs+8fnyLvooQfR5iujAnR9bxfY2xOVOxFS8SK3Le0l48VyYu1M8HRe5JD8wKPTjYnifaK3Wfn/GChYQ8ZAi6WRzWgqLV5YrsVLnZaVSoXU1g9gOIDwFySiGi+Zdrnzr7J3r+SMuszlcQCRn8lNGcTuSy2jOI7o9mxjZo+vR3ej3tN+ifRSOyUTS0+VMOid93cCubeiy/6TImS0QxRSCq2vxKr45zV+FQnjWH6D2xg+E9EatLcLAdHTgtGGD80D6jM0+aOl4wJgO/f96R2aJKCQ3yvgftRhdFMOpd6oAAAAASUVORK5CYII=" role="presentation" style="pointer-events: none;" width="16" height="16">
            </div>
        </div>
        <div class="js-toggle-thumb"></div>
        <input class="js-toggle-screenreader-only" type="checkbox" aria-label="Switch between Dark and Light mode">
    </div>
</div>

<style>
  .js-toggle-wrapper {
    display: table;
    margin: 0 auto;
    position:relative;
    top:1em
  }

  .js-toggle {
    touch-action: pan-x;
    display: inline-block;
    position: relative;
    cursor: pointer;
    background-color: transparent;
    border: 0;
    padding: 0;
    -webkit-touch-callout: none;
    user-select: none;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
    -webkit-tap-highlight-color: transparent;
  }
  
  .js-toggle-screenreader-only {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }
  
  .js-toggle-track {
    width: 50px;
    height: 24px;
    padding: 0;
    border-radius: 30px;
    background-color: hsl(222, 14%, 7%);
    transition: all 0.0s ease;
  }
  
  .js-toggle-track-check {
    position: absolute;
    width: 17px;
    height: 17px;
    left: 5px;
    top: 0px;
    bottom: 0px;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    opacity: 0;
    transition: opacity 0.0s ease;
  }
  
  .js-toggle--checked .js-toggle-track-check {
    opacity: 1;
    transition: opacity 0.0s ease;
  }
  
  .js-toggle-track-x {
    position: absolute;
    width: 17px;
    height: 17px;
    right: 5px;
    top: 0px;
    bottom: 0px;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    opacity: 1;
    transition: opacity 0.0s ease;
  }
  
  .js-toggle--checked .js-toggle-track-x {
    opacity: 0;
  }
  
  .js-toggle-thumb {
    position: absolute;
    top: 1px;
    left: 1px;
    width: 22px;
    height: 22px;
    border-radius: 50%;
    background-color: #fafafa;
    box-sizing: border-box;
    transition: all 0.0s cubic-bezier(0.23, 1, 0.32, 1) 0ms;
    transform: translateX(0);
  }
  
  .js-toggle--checked .js-toggle-thumb {
    transform: translateX(26px);
    border-color: #19ab27;
  }
  
  .js-toggle--focus .js-toggle-thumb {
    box-shadow: 0px 0px 2px 3px rgb(180, 180, 180);
  }
  
  .js-toggle:active .js-toggle-thumb {
    box-shadow: 0px 0px 5px 5px rgb(180, 180, 180);
  }

  body.dark-mode, 
  body.dark-mode body * {
    background: #1b1b1b;
    color: navajowhite;
  }

  body.dark-mode h1 { color: darksalmon; }
  body.dark-mode h2 { color: #C5908E; }
  body.dark-mode h3 { color: #C2E5D3; }
  body.dark-mode h4 { color: #D5D6EA; }
  body.dark-mode h5, body.dark-mode h6 { color: #EBDDE2; }
  body.dark-mode a { color: thistle; }
  body.dark-mode a:visited { color: #F7CAC9; }
</style>

<script>
  function getStyleSheet(file_name) {
    for (var i = 0; i < document.styleSheets.length; i++) {
      var sheet = document.styleSheets[i];
      if (sheet.href.includes(file_name)) {
        return sheet;
      }
    }
  }
  var body = document.body;
  var switcher = document.getElementsByClassName('js-toggle')[0];
  switcher.addEventListener("click", function() {
    this.classList.toggle('js-toggle--checked');
    this.classList.add('js-toggle--focus');
    if (this.classList.contains('js-toggle--checked')) {
      body.classList.add('dark-mode');
      localStorage.setItem('darkMode', 'true');
      getStyleSheet("syntax_dark").disabled = false;
      getStyleSheet("syntax_light").disabled = true;
    } else {
      body.classList.remove('dark-mode');
      setTimeout(function() {localStorage.removeItem('darkMode');}, 100);
      getStyleSheet("syntax_dark").disabled = true;
      getStyleSheet("syntax_light").disabled = false;
    }
  })
  if (localStorage.getItem('darkMode')) {
    switcher.classList.add('js-toggle--checked');
    body.classList.add('dark-mode');
    getStyleSheet("syntax_dark").disabled = false;
    getStyleSheet("syntax_light").disabled = true;
  } else {
    getStyleSheet("syntax_dark").disabled = true;
    getStyleSheet("syntax_light").disabled = false;
  }
</script>

  


<body>
<nav>
    <ul>
      <li style="display: inline-block; margin-right: 10px;"><a href="/" style="font-size: 25px; text-decoration: none;"><span class=pre></span><span>Home</span></a></li><li style="display: inline-block; margin-right: 10px;"><a href="/publications/" style="font-size: 25px; text-decoration: none;"><span class=pre></span><span>Publications</span></a></li><li style="display: inline-block; margin-right: 10px;"><a href="/research/" style="font-size: 25px; text-decoration: none;"><span class=pre></span><span>Research</span></a></li>
    </ul>
</nav>
<main>
<header><h1 id="tag - Binning mitochondrial reads from skimming data using krepp">Binning mitochondrial reads from skimming data using krepp</h1></header>

<hr>
<article>
<h2 id="installing-krepp">Installing krepp</h2>
<p>We start by installing krepp, the easiest way is through <a href="https://anaconda.org/bioconda/krepp">bioconda</a>.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">conda install bioconda::krepp
</span></span><span class="line"><span class="ln">2</span><span class="cl">krepp --help
</span></span></code></pre></div><p>To perform any kind of query, you need to either build an index from reference genomes or download a public krepp index from one of our servers.
It is quite straightforward to download an index and get started (jump to <a href="#downloading-an-existing-index">Downloading an existing index</a>).
Alternatively, if you have your own reference mitogenomes, you can index them from scratch (see <a href="#indexing-mitogenomes">Indexing mitogenomes</a>).
Once you do either of these, the next step is querying your skimming data (e.g., a FASTA or FASTQ file), as we will discuss in detail.</p>
<h2 id="downloading-an-existing-index">Downloading an existing index</h2>
<p>Download a relatively up-to-date mitogenome index by running the below <code>wget</code> command. Then, untar and decompress it.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">wget --no-certificate https://ter-trees.ucsd.edu/data/krepp/index_mitochondrion-RefSeq20240913.tar.gz
</span></span><span class="line"><span class="ln">2</span><span class="cl">tar -xzvf index_mitochondrion-RefSeq20240913.tar.gz
</span></span></code></pre></div><p>The resulting directory <code>index_mitochondrion-RefSeq20240913</code> stores the index that we can perform queries against.
You can jump to the <a href="#querying-your-skimming-data">next step</a>.</p>
<h2 id="indexing-mitogenomes">Indexing mitogenomes</h2>
<p>For example, we will use a recent RefSeq release of all available mitochondrial genomes.
The very first step is to download all the mitogenomes available on RefSeq as of 2024/09/13, and decompress the downloaded file.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl"><span class="c1"># 2024-09-13</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">wget https://ftp.ncbi.nlm.nih.gov/refseq/release/mitochondrion/mitochondrion.1.1.genomic.fna.gz
</span></span><span class="line"><span class="ln">3</span><span class="cl">gunzip -d mitochondrion.1.1.genomic.fna.gz
</span></span></code></pre></div><p>Convert multi-line FASTA to 2-line FASTA for easier processing.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">seqtk seq -l0 mitochondrion.1.1.genomic.fna &gt; /tmp/mitochondrion_l0.1.1.genomic.fna <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="se"></span>  mv /tmp/mitochondrion_l0.1.1.genomic.fna mitochondrion.1.1.genomic.fna
</span></span></code></pre></div><p>Since krepp requires one reference sequence per file, we have to split concatenated mitogenomes into separate files:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">mkdir mitochondrion_genomic_peracc
</span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="nb">cd</span> mitochondrion_genomic_peracc
</span></span><span class="line"><span class="ln">3</span><span class="cl">cat ../mitochondrion.1.1.genomic.fna <span class="se">\
</span></span></span><span class="line"><span class="ln">4</span><span class="cl"><span class="se"></span>  <span class="p">|</span> sed <span class="s1">&#39;s/^&gt;/&gt;&gt;/&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="ln">5</span><span class="cl"><span class="se"></span>  <span class="p">|</span> awk <span class="s1">&#39;BEGIN{RS=&#34;&gt;&gt;&#34;;FS=&#34;\n&#34;} NR&gt;1{split($0,acc,&#34; &#34;); fnme=acc[1]&#34;.fasta&#34;; print &#34;&gt;&#34; $0 &gt; fnme; close(fnme);}&#39;</span>
</span></span><span class="line"><span class="ln">6</span><span class="cl"><span class="nb">cd</span> ../
</span></span></code></pre></div><p>Then, list all sequence identifiers in another file and create a mapping between accession IDs and each file corresponding to a reference sequence.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">grep <span class="s2">&#34;^&gt;&#34;</span> mitochondrion.1.1.genomic.fna <span class="p">|</span> sed <span class="s1">&#39;s/^&gt;//&#39;</span> &gt; seqids_mitochondrion.txt
</span></span><span class="line"><span class="ln">2</span><span class="cl">cut -f1 -d<span class="s1">&#39; &#39;</span> seqids_mitochondrion.txt <span class="p">|</span> awk <span class="s1">&#39;{print $1&#34;\t./mitochondrion_genomic_peracc/&#34;$2&#34;.fasta&#34;;}&#39;</span> &gt; input_map-mitochondrion.tsv
</span></span></code></pre></div><p>This file (<code>input_map-mitochondrion.tsv</code>) is going to be used as the input to krepp.</p>
<p>We are almost there. Now that we have a mapping, we can index mitogenomes by simply running the following <code>krepp index</code> command.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">seq <span class="m">0</span> <span class="m">7</span> <span class="p">|</span> <span class="se">\
</span></span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="se"></span>    xargs -I<span class="o">{}</span> bash -c <span class="s2">&#34;krepp index \
</span></span></span><span class="line"><span class="ln">3</span><span class="cl"><span class="s2">    -i input_map-mitochondrion.tsv -o index_mitochondrion-RefSeq20240913 \
</span></span></span><span class="line"><span class="ln">4</span><span class="cl"><span class="s2">    -k 29 -w 32 -h 13 -m 8 -r {} --no-frac --sdust-t 0 --num-threads 24&#34;</span>
</span></span></code></pre></div><p>Here, <code>index_mitochondrion-RefSeq20240913</code> is the name of the directory where we store the index.
For all other parameters, you can refer to the tutorial in the GitHub repository.</p>
<h2 id="querying-your-skimming-data">Querying your skimming data</h2>
<p>We will use the <code>dist</code> subcommand of krepp via <code>krepp dist</code>.
Suppose you have a FASTQ file named <code>xyz.fq</code> and an index in a directory <code>index_mitochondrion-RefSeq20240913</code>.
To compute the distance between each read in <code>xyz.fq</code> and all sufficiently close mitogenomes indexed, run the command below.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl"><span class="c1"># Compute distances and filter any mapping above 15% sequence similarity</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">krepp dist -i index_mitochondrion-RefSeq20240913 -q xyz.fq --dist-max 0.15 --filter
</span></span></code></pre></div><p>This will output estimated distances to stdout in a tab-separated format.
You can save this to a file with the <code>-o</code> option (e.g., add <code>-o xyz-distances.tsv</code> to the command above).
The first 10 lines would look something like this.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl"><span class="c1"># Print distances to stdout and get the first 10 lines</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">krepp dist -i index_mitochondrion-RefSeq20240913 -q xyz.fq --dist-max 0.15 --filter <span class="p">|</span> head
</span></span></code></pre></div><p>Output:</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-fallback" data-lang="fallback"><span class="line"><span class="ln">1</span><span class="cl">#software: krepp  #version: v0.5.1  #invocation :krepp dist -i index_mitochondrion-RefSeq20240913 -q xyz.fq --dist-max 0.15 --filter
</span></span><span class="line"><span class="ln">2</span><span class="cl">SEQ_ID	REFERENCE_NAME	DIST
</span></span><span class="line"><span class="ln">3</span><span class="cl">AG-359-G18_1-1273	NaN	NaN
</span></span><span class="line"><span class="ln">4</span><span class="cl">AG-359-G18_1-1270	NC_065012.1 Amynthas deogyusanensis mitochondrion	2.00371e-05
</span></span><span class="line"><span class="ln">5</span><span class="cl">AG-359-G18_1-1272	NaN	NaN
</span></span><span class="line"><span class="ln">6</span><span class="cl">AG-359-G18_1-1269	NC_065012.1 Amynthas deogyusanensis mitochondrion	0.0137122
</span></span><span class="line"><span class="ln">7</span><span class="cl">AG-359-G18_1-1268	NC_065012.1 Amynthas deogyusanensis mitochondrion	0.00638889
</span></span><span class="line"><span class="ln">8</span><span class="cl">AG-359-G18_1-1267	NC_065012.1 Amynthas deogyusanensis mitochondrion	0.0163656
</span></span><span class="line"><span class="ln">9</span><span class="cl">AG-359-G18_1-1271	NaN	NaN
</span></span></code></pre></div><p>The first column is for the read ID, the second column is for the accession ID of the references the corresponding read maps to, and the third column is for the distances between the reads and the mapped references.
Notice that we used a threshold of 15% sequence similarity (<code>--dist-max 0.15</code>).
This means that any read without a hit below 15% similarity will have <code>NaN</code> in the second and third columns.
These are most likely non-mitochondrial reads (e.g., bacterial).
You can simply filter such reads using their IDs and move forward with the rest.
You may want to change the threshold value (e.g., <code>--dist-max 0.1</code> for a more aggressive filtering or <code>--dist-max 0.2</code> for a more conservative filtering), but 15% should do a decent job for this purpose.</p>
<p>To test these commands, I created a toy example of a skimming file consisting of 500 mitochondrial reads (generated from a randomly selected set of mitogenomes) and 500 bacterial reads.
When I run the below command, it outputs 500 mitochondrial read IDs, meaning that we filter all bacterial reads and retain all mitochondrial reads, which is the desired output.</p>
<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="ln">1</span><span class="cl">krepp dist -i index_mitochondrion-RefSeq20240913 -q xyz.fq --dist-max 0.15 --filter <span class="se">\
</span></span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="se"></span> <span class="p">|</span> grep -v NaN <span class="p">|</span> cut -f1 <span class="p">|</span> uniq
</span></span></code></pre></div><p>Good luck with your data!</p>


<div id="nextprev">
    <a href="/posts/krank-mapping-baits/"><div id="prevart">Previous:<br>Mapping mitochondrial baits to taxonomic groups using KRANK</div></a>
</div>
<br clear=both><div style="clear:both" class=taglist>Related<br><a id="tag_k-mers" href="https://bo1929.github.io/tags/k-mers">k-mers</a> &middot; <a id="tag_metagenomics" href="https://bo1929.github.io/tags/metagenomics">metagenomics</a> &middot; <a id="tag_mitochondrial-reads" href="https://bo1929.github.io/tags/mitochondrial-reads">mitochondrial-reads</a> &middot; <a id="tag_distance-estimation" href="https://bo1929.github.io/tags/distance-estimation">distance-estimation</a> &middot; <a id="tag_krepp" href="https://bo1929.github.io/tags/krepp">krepp</a> &middot; <a id="tag_genome-skimming" href="https://bo1929.github.io/tags/genome-skimming">genome-skimming</a> &middot; <a id="tag_read-mapping" href="https://bo1929.github.io/tags/read-mapping">read-mapping</a></div>
</article>

<hr>
</main>


<footer>
	<a href="https://bo1929.github.io/">https://bo1929.github.io/</a><br><br><a href="/index.xml"><img src="/rss.svg" style="max-height:1.5em" alt="RSS Feed" title="Subscribe via RSS for updates."></a>
</footer>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript"
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

</body>

</html>
