<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>CONSULT-II won the best paper award - Ali Osman Berk Şapcı</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="Ali Osman Berk Şapcı" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">Ali Osman Berk Şapcı</div>
					<div class="logo__tagline">Personal website for news, notes, and maybe bl*g posts.</div>
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">CONSULT-II won the best paper award</h1>
			
		</header>
		<div class="content post__content clearfix">
			<p>I presented at RECOMB Comparative Genomics 2023, in Istanbul (see my <a href="/presentations/consultii-recombcg-presentation-2023.pdf">slides</a>).
Our paper, with Elenora Rachtman and Siavash Mirarab, <a href="https://www.doi.org/10.1007/978-3-031-36911-7_13">CONSULT-II: Taxonomic Identification Using Locality Sensitive Hashing</a> won the best paper award.</p>
<p>The abstract is given below.</p>
<blockquote>
<p>Metagenomics is widely used to study the microbiome using environmental samples, and taxonomic classification of reads is a precursor to many analyses of such data. Taxonomic classification requires comparing sample reads against a reference dataset of known organisms. Crucially, the genomes represented in a sample may be phylogenetically distant from their closest match in the reference set. Thus, simply mapping reads to genomes is insufficient; we need to find inexact matches to species with substantial distance. While k-mer-based methods, such as Kraken, have proved popular, they have limited ability to match against distant taxa. In this paper, we use locality sensitive hashing to design a k-mer-based method that can match reads to genomes with higher distance than existing methods. We build on an earlier contamination detection method, CONSULT, to add taxonomic classification abilities. We show in a series of experiments that our method, CONSULT-II, has higher recall than alternatives when precision is about the same. Its results can also be summarized to obtain a taxonomic profile, which we show outperforms leading methods with respect to some measurement criteria. CONSULT-II is available at <a href="https://github.com/bo1929/CONSULT-II">https://github.com/bo1929/CONSULT-II</a>.</p>
</blockquote>
<figure><img src="/images/me-presenting-at-recombcg-2023.jpg"/><figcaption>
            <h4>A shot taken by Can Alkan during the presentation, İTÜ Maçka Oteli, Istanbul.</h4>
        </figcaption>
</figure>
		</div>
	</article>
</main>




			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2024 Ali Osman Berk Şapcı.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>